hector_pose_prediction_ros/include/hector_pose_prediction_ros/visualization.h

```diff
@@ -0,0 +1,230 @@
+#ifndef HECTOR_POSE_PREDICTION_ROS_VISUALIZATION_H
+#define HECTOR_POSE_PREDICTION_ROS_VISUALIZATION_H
+
+#include <hector_math/math/operations.h>
+#include <hector_math/types.h>
+#include <hector_math_ros/message_conversions/geometry_msgs.h>
+#include <hector_pose_prediction_interface/types.h>
+#include <std_msgs/ColorRGBA.h>
+#include <visualization_msgs/MarkerArray.h>
+
+namespace hector_pose_prediction_interface
+{
+namespace visualization
+{
+template<typename Scalar>
+void addSupportPolygonEdgesToMarkerArray( visualization_msgs::MarkerArray &marker_array,
+                                          const hector_math::Vector3List<Scalar> &contact_hull_points,
+                                          const std::vector<std_msgs::ColorRGBA> &edge_colors,
+                                          const std::string &frame_id, const std::string &ns )
+{
+  marker_array.markers.reserve( marker_array.markers.size() + contact_hull_points.size() );
```

Comment:
You could add an assert here that `edge_colors` and `contact_hull_points` have the same length.
Or also throw an exception.

Reply:
Good point, somehow I missed it there.